applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_structural_response_function.h

```diff
@@ -0,0 +1,225 @@
+                                              const Matrix& rDerivativesMatrix,
+                                              Vector& rResponseGradient,
+                                              ProcessInfo& rProcessInfo);
+
+    virtual void CalculateSensitivityGradient(Element& rAdjointElem, 
+                                              const Variable<array_1d<double,3>>& rVariable,
+                                              const Matrix& rDerivativesMatrix,
+                                              Vector& rResponseGradient,
+                                              ProcessInfo& rProcessInfo);
+
+    virtual void CalculateSensitivityGradient(Condition& rAdjointCondition, 
+                                              const Variable<array_1d<double,3>>& rVariable,
+                                              const Matrix& rDerivativesMatrix,
+                                              Vector& rResponseGradient,
+                                              ProcessInfo& rProcessInfo);
+
+    void AssembleNodalSensitivityContribution(Variable<double> const& rSensitivityVariable,
+                                              Vector const& rSensitivityVector,
+                                              Element::GeometryType& rGeom);
+
+    void AssembleNodalSensitivityContribution(Variable<array_1d<double, 3>> const& rSensitivityVariable,
+                                              Vector const& rSensitivityVector,
+                                              Element::GeometryType& rGeom);
+
+    void AssembleElementSensitivityContribution(Variable<double> const& rSensitivityVariable,
+                                                Vector const& rSensitivityVector,
+                                                Element& rElem);
+
+    void AssembleElementSensitivityContribution(Variable<array_1d<double, 3>> const& rSensitivityVariable,
+                                                Vector const& rSensitivityVector,
+                                                Element& rElem);
+
+    void AssembleConditionSensitivityContribution(Variable<double> const& rSensitivityVariable,
+                                              Vector const& rSensitivityVector,
+                                              Element::GeometryType& rGeom);
+
+    void AssembleConditionSensitivityContribution(Variable<array_1d<double, 3>> const& rSensitivityVariable,
+                                              Vector const& rSensitivityVector,
+                                              Element::GeometryType& rGeom);
+
+    void ReadDesignVariables(std::vector<std::vector<Variable<double>>>& rScalarDesignVariables, 
+        std::vector<std::vector<Variable<array_1d<double,3>>>>& rVectorDesignVariables, Parameters DesignVariableSettings); 
+    ///@}
+
+private:
+    ///@name Member Variables
+    ///@{
+
+    std::string mSensitivityModelPartName;
```

Comment:
Serialization

Reply:
@philbucher is serialization really needed for the response functions? Also in core response base class developed within #2423 by @msandre no serialization is used.